### PR TITLE
Implement database encryption with SQLCipher

### PR DESCRIPTION
**Pull request by pasin**
- Integrated with couchbase-lite-java-sqlcipher instead of couchbase-lite-java-native.
- Removed CollationTest as it should be moved to couchbase-lite-java-sqlcipher instead (https://github.com/couchbaselabs/couchbase-lite-java-sqlcipher/issues/1).
- Rewritten JavaSQLiteStorageEngine to use SQLiteDatabase from couchbase-lite-java-sqlcipher

Reference: https://github.com/couchbase/couchbase-lite-java/issues/68

### CODE REVIEW

File: src/main/java/com/couchbase/lite/storage/JavaSQLiteStorageEngine.java [hunk 1]
```diff
@@ -1,514 +1,289 @@
...
+                database.close();
+            throw new SQLException(e);
+        }
+        return database.isOpen();
     }
 
-    private native int _getVersion(long handle);
-
     @Override
     public int getVersion() {
```
Comment (hideki) [inline src/main/java/com/couchbase/lite/storage/JavaSQLiteStorageEngine.java]: This is general comments for all methods that are declared in this class.
- As all method could throw SQLException, but method signature is not defined with throws SQLException. Is it correct?

**Comment by hideki** [comment]
Once question about SQLException is answered, this PR is ready to be merged.

**Comment by pasin** [comment]
I made changes per your comment. Please review.

**Comment by pasin** [comment]
@hideki I'm revising the throws statement. Please don't review my previous fix yet.

**Pull request merged**